keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,532 @@
+    checks on both Endpoints and EndpointSlices. This will need to be
+    split into separate tests, with the Endpoints tests
+    feature-tagged.
+
+  - Some of the tests in `test/e2e/network/endpointslice.go` will need
+    to be split up, to separately test "EndpointSlices are created
+    correctly" and "EndpointSlices match Endpoints" in separate tests,
+    with the Endpoints tests feature-tagged.
+
+  - The conformance tests in
+    `test/e2e/network/endpointslicemirroring.go` should be demoted
+    from conformance, and all of the tests should be feature-tagged,
+    but should otherwise be unchanged.
+
+  - Several tests in `test/e2e/network/dual_stack.go` check that the
+    Endpoints controller does the right thing but _do not_ check that
+    the EndpointSlice controller does the right thing (which means
+    that we do not actually have any proper e2e testing of dual-stack
+    Services). These should be updated to test EndpointSlices, with
+    the Endpoints tests split out into separate feature-tagged tests.
+
+  - `[It] [sig-network] Services should test the lifecycle of an
+    Endpoint [Conformance]`: This just tests that the API works and
+    does not test the behavior of the controllers, so it just needs to
+    be feature-tagged, but not changed.
+
+  - `[It] [sig-network] Service endpoints latency should not be very
+    high [Conformance]`: This tests the latency of the Endpoints
+    controller, and should be fixed to test the latency of the
+    EndpointSlice controller instead, since the latency of the
+    Endpoints controller has no impact on the functioning of a
+    cluster.
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Alpha
+
+- All Endpoints and EndpointSlice Mirroring tests have been demoted
+  from conformance, and we document this (e.g. via a blog post).
+
+- All official Kubernetes documentation is updated to primarily
+  discuss EndpointSlices, and to mention Endpoints only as a
+  deprecated API. No examples use `kubectl get endpoints` or involve
+  creating an Endpoints object.
+
+#### Beta
+
+- (There is no requirement for any particular amount of time to have
+  passed between Alpha and Beta.)
+
+- We have a passing "no Endpoints" e2e job.
+
+- Warnings are emitted when users create Endpoints.
+
+- If we are going to add metrics or other "Am I using Endpoints?"
+  diagnostics, then they are added and have tests.
+
+- The documentation explains how to disable the Endpoints and
+  EndpointSlice Mirroring controllers, but warns that third-party
+  components may still depend on them.
```

Comment:
Bear in mind that in managed Kubernetes, people who use the cluster may have no way to disable these controllers.